go/arrow/flight/flightsql/client.go

```diff
@@ -518,22 +544,44 @@ func (p *PreparedStatement) GetSchema(ctx context.Context) (*flight.SchemaResult
 	return p.client.getSchema(ctx, desc, p.opts...)
 }
 
-// SetParameters takes a record batch to send as the parameter bindings when
-// executing. It should match the schema from ParameterSchema.
-//
-// This will call Retain on the record to ensure it doesn't get released
-// out from under the statement. Release will be called on a previous
-// binding record if it existed, and will be called upon calling Close
-// on the PreparedStatement.
-func (p *PreparedStatement) SetParameters(binding arrow.Record) {
+func (p *PreparedStatement) clearParameters() {
 	if p.paramBinding != nil {
 		p.paramBinding.Release()
 		p.paramBinding = nil
 	}
+	if p.streamBinding != nil {
+		p.streamBinding.Release()
+		p.streamBinding = nil
+	}
+}
+
+// SetParameters takes a record batch to send as the parameter bindings when
+// executing. It should match the schema from ParameterSchema.
+//
+// This will call Retain on the record to ensure it doesn't get released out
+// from under the statement. Release will be called on a previous binding
+// record or reader if it existed, and will be called upon calling Close on the
+// PreparedStatement.
+func (p *PreparedStatement) SetParameters(binding arrow.Record) {
+	p.clearParameters()
 	p.paramBinding = binding
 	p.paramBinding.Retain()
 }
```

Comment:
should probably mention in the docstring that it will clear any existing bound params

Reply:
nvm just saw you have `Release will be called on a previous binding record or reader`, ignore this :)